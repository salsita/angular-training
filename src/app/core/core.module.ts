import { HttpClientModule } from '@angular/common/http';
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { ApiModule } from '@salsita/ng-api';
import { CrudModule } from '@salsita/ng-crud';
import { EntityRepositoryModule } from '@salsita/ng-entities';
import { RouterModule as RouterStoreModule } from '@salsita/ng-router';

import { environment } from '../../environments/environment';
import { SharedModule } from '../shared/shared.module';
import { ApiErrorToastComponent } from './api/api-error-toast/api-error-toast.component';
import { ApiLoaderComponent } from './api/api-loader/api-loader.component';

@NgModule({
  imports: [
    HttpClientModule,

    ApiModule.forRoot({ storeNamespace: 'api', baseUrl: environment.apiBaseUrl }),
    CrudModule.forRoot({ storeNamespace: 'crud' }),
    EntityRepositoryModule.forRoot({ storeNamespace: 'entityRepository' }),
    RouterStoreModule.forRoot({ storeNamespace: 'router' }),
    SharedModule
  ],
  declarations: [ApiErrorToastComponent, ApiLoaderComponent],
  exports: [ApiErrorToastComponent, ApiLoaderComponent, ApiModule, HttpClientModule]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        `CoreModule has already been loaded. Import Core modules in the AppModule only.`
      );
    }
  }
}
