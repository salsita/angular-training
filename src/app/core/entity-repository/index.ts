export {
  EntityRepositoryModuleConfig,
  EntityRepositoryState,
  EntityRepositoryStateRoot,
  SingleEntityRepository
} from './entity-repository.interfaces';
export { EntityRepositorySelectors } from './entity-repository.selectors';
export { EntityRepositoryService } from './entity-repository.service';
export { EntityRepositoryModule } from './entity-repository.module';
