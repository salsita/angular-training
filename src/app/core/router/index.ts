export {
  FlatRouterStateSnapshot,
  RouterModuleConfig,
  RouterState,
  RouterStateRoot
} from './router.interfaces';
export { RouterSelectors } from './router.selectors';
export { CustomRouterStateSerializer } from './router.serializer';
export { RouterModule } from './router.module';
