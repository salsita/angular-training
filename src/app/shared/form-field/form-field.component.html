<label>{{label}}</label>
<input
  [formControl]="control"
  [type]="type"
  [placeholder]="label"
/>
<span *ngIf="control && control.touched && control.errors">
  <ng-container *ngFor="let error of errors; let i = index">
    <ng-container *ngIf="i > 0">,</ng-container>
    {{error}}
  </ng-container>
</span>
