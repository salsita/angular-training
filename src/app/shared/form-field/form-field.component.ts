import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'app-form-field',
  templateUrl: './form-field.component.html',
  styleUrls: ['./form-field.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class FormFieldComponent {
  @Input() label = '';
  @Input() type = 'text';
  @Input() control: FormControl = null;

  get errors() {
    return this.control && this.control.errors && Object.keys(this.control.errors);
  }
}
