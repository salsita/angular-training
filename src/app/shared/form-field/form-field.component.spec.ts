import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormControl, ReactiveFormsModule } from '@angular/forms';

import { FormFieldComponent } from './form-field.component';

describe('FormFieldComponent', () => {
  let component: FormFieldComponent;
  let fixture: ComponentFixture<FormFieldComponent>;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [ReactiveFormsModule],
        declarations: [FormFieldComponent]
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(FormFieldComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    component.label = 'Label';
    component.control = new FormControl();
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
