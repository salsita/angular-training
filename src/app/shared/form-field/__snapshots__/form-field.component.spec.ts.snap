// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldComponent should create 1`] = `
<app-form-field
  control={[Function FormControl]}
  label={[Function String]}
  type={[Function String]}
>
  <label>
    Label
  </label>
  <input
    class="ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
    placeholder="Label"
    type="text"
  />
  
  
</app-form-field>
`;
